internal/coreinternal/processor/filterset/regexp/regexpfilterset.go

```diff
@@ -41,7 +43,17 @@ func NewFilterSet(filters []string, cfg *Config) (*FilterSet, error) {
 
 	if cfg != nil && cfg.CacheEnabled {
 		fs.cacheEnabled = true
-		fs.cache = lru.New(cfg.CacheMaxNumEntries)
+		var err error
+		if cfg.CacheMaxNumEntries == 0 {
```

Comment:
I was thinking of disallowing `0` value and setting `512` as default. Not sure if it's good to associate `0` default value with `512`, at it should be documented